src/file.ts

```diff
@@ -1752,12 +1754,20 @@ class File extends ServiceObject<File> {
       md5,
     });
 
+    const progressStream = new ProgressStream();
+    if (options.onUploadProgress) {
+      progressStream.on('progress', bytesRead => {
+        options.onUploadProgress!({bytesRead});
```

Comment:
[previous comment](https://github.com/AVaksman/nodejs-storage/pull/12#discussion_r364464162)

We should just forward this event to the returned stream, rather than the handler (IMO). Keeps with the streaming API, so we don't have to mix callbacks/streams.

From `bucket#upload`, it should accept `onUploadProgress` there:

```js
Bucket.prototype.upload = function (opts) {
  file.createWriteStream()
    .on('progress', opts.onUploadProgress)
}
```

But I don't believe `file#createWriteStream` should accept the option itself.

Reply:
with new changes `bucket#upload` and `file#save` accept `onUploadProgress`